### PR TITLE
`Orchestrator.RunnableProjects` tests refactoring

### PR DESCRIPTION
**Pull request by vlada-shubina**
### Problem
while working on new tests I discovered that there almost 2 identical classes in `TestHelper` and invalid `TestTemplateSetup` implementation.

### Solution
- unified classes in `TestHelper`, documented them
- removed `TestTemplateSetup`
- partially implemented `TemplateConfigModel.ToJObject` method for JSON serialization required in tests

### Checks:
- [ ] Added unit tests
- [x] Added `#nullable enable` to all the modified files [?](https://github.com/dotnet/templating/wiki/Contributing#coding-style) - enabled project-wide in `Orchestrator.RunnableProjects.UnitTests`

### CODE REVIEW

File: src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/ConfigModel/TemplateConfigModel.cs [hunk 1]
```diff
@@ -507,7 +515,358 @@ private static IReadOnlyList<BindSymbol> SetupImplicitBindSymbols()
...
+                        {
+                            continue;
+                        }
+                        serializer.Serialize(writer, p);
+                    }
+                    writer.WriteEndObject();
+
+                    if (value.Symbols.Any(s => s is not ParameterSymbol && !s.IsImplicit))
+                    {
+                        throw new NotSupportedException("Symbols are not supported for serialization to JSON.");
```
Comment (JanKrivanek) [inline src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/ConfigModel/TemplateConfigModel.cs:847]: Message seams bit missleading. Maybe "Only Implicit and Parameter symbols are supported for serialization to JSON."?

Comment (JanKrivanek) [reply src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/ConfigModel/TemplateConfigModel.cs:847]: Just wondering - why are the other types not supported? Will wee need the serializer when validating e.g. template with calculated symbols?

**Review by JanKrivanek — APPROVED** [review]
Looks good - I like the unification and simplification of test helpers.
I'm not sure if I fully understand the usage scenarios behind `TemplateConfigModelJsonConverter` and why other symbol types are not supported.

But overall - we should merge to simplify review of #5329

File: src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/ConfigModel/TemplateConfigModel.cs (hunk 1 above)
Comment (vlada-shubina) [reply src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/ConfigModel/TemplateConfigModel.cs:847]: It just needs more work and not required for time being. I'll create an issue for this. Also I will try to move this serialization to tests not to misuse unintentionally.

Comment (vlada-shubina) [reply src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/ConfigModel/TemplateConfigModel.cs]: Moved serialization to tests as it is not complete and can be used only for tests.
The idea is that if the next tests need more data, the serialization may be implemented then.

**Pull request merged**